compiler/include/codegen.h

```diff
@@ -78,6 +91,8 @@ struct GenInfo {
   llvm::IRBuilder<> *builder;
   LayeredValueTable *lvt;
 
+  std::stack<LoopData> loopStack;
```

Comment:
Since this stack only ever contains elements with parallel=true and since you're not extending other loop generation (while loops, other loops, ...), I think this stack should just store llvm::MDNode* for parallel loops and be called something like `parallelLoopStack`.
Do you envision some case in which this stack will need to contain non-parallel loops?